scrypto/src/resource/non_fungible_id.rs

```diff
@@ -66,8 +66,8 @@ impl TryFrom<&[u8]> for NonFungibleId {
     type Error = ParseNonFungibleIdError;
 
     fn try_from(slice: &[u8]) -> Result<Self, Self::Error> {
-        let value = ScryptoValue::from_slice_no_custom_values(slice)
-            .map_err(|_| ParseNonFungibleIdError::InvalidValue)?;
+        let value =
+            ScryptoValue::from_slice(slice).map_err(|_| ParseNonFungibleIdError::InvalidValue)?;
```

Comment:
Instead of using `from_slice` can we modify `from_slice_no_custom_values` to include `Decimal`. This is mainly so that we don't accept buckets/vaults/etc.

Reply:
Thought about it, this is non-urgent and might be better to wait until that becomes an Enum to add that in with that refactor.